server/server.js

```diff
@@ -2,6 +2,10 @@ const jsonServer = require('json-server');
 const server = jsonServer.create();
 const router = jsonServer.router('db.json');
 const middlewares = jsonServer.defaults();
+const fs = require('fs');
+
+
+const data = JSON.parse(fs.readFileSync('./db.json', { encoding: 'utf8' }));
```

Comment:
why you use synk way?

Reply:
If we use async function here, we can't be sure that data has been loaded and parsed when we get a request, since the server will start running before the execution of async function. Using a sync function ensures, that the data has been loaded by the time we declare a route for getting that data

Reply:
If you have questions regarding the event loop in JavaScript, I can recommend this excellent article: https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/. It helped me a lot when I was researching the matter. You can also find a video on the youtube by the same speaker (Jake Archibald) on the topic

Reply:
Thanks! But this article covers event loop which works on client side, but Nodejs uses libuv (a C library that implements the Node.js event loop and all asynchronous platform operations. 
The “normal” way to read files in Node.js is to read asynchronously. This means that you call the file reading command and send a callback, which will be called when the reading is completed. This allows you to work with multiple read requests in parallel.
this code works perfectly:

const fs = require('fs');
let db;
fs.readFile('./db.json', 'utf8', function(err, content) {
    db = content;
});

server.use((req, res, next) => {
    const data = JSON.parse(db);

Reply:
You are absolutely right, but there is a chance that db will be undefined if request comes sooner than Node finishes reading file. We could in theory read file asynchronously and add following line in your example:
```javascript
if (!db) {
  return;
}
```
But that will put unnecessary load on server once it's running.

Reply:
I've done some more testing and it looks like db.json has to be absolutely massive in order for time window between server starting listening and file loading to be any noticeable. And surprisingly to me, it doesn't crash the server, just sends 500 error. I can change it to async if you insist.

Reply:
Ok. Great improvement.

Reply:
Fixed